frontend/www/js/omegaup/markdown.ts

```diff
@@ -398,19 +398,28 @@ export class Converter {
               .replace(/</g, '&lt;')
               .replace(/>/g, '&gt;');
           }
-          if (indentation != '') {
-            const lines = [];
-            const stripPrefix = new RegExp('^ {0,' + indentation.length + '}');
-            for (const line of contents.split('\n')) {
-              lines.push(line.replace(stripPrefix, ''));
-            }
+
+          if (indentation == '') {
             contents = escapeCharacters(
-              lines.join('\n'),
+              contents,
               ' \t*_{}[]()<>#+=.!|`-',
               /*afterBackslash=*/ false,
               /*doNotEscapeTildeAnDollar=*/ true,
             );
+            return `<pre><code${className}>${contents}</code></pre>`;
+          }
+
+          const lines = [];
+          const stripPrefix = new RegExp('^ {0,' + indentation.length + '}');
+          for (const line of contents.split('\n')) {
+            lines.push(line.replace(stripPrefix, ''));
           }
+          contents = escapeCharacters(
+            lines.join('\n'),
+            ' \t*_{}[]()<>#+=.!|`-',
+            /*afterBackslash=*/ false,
+            /*doNotEscapeTildeAnDollar=*/ true,
+          );
```

Comment:
parece que esto se puede estructurar mejor como

```suggestion
          if (indentation !== '') {
            // Delete any extra indentation spaces from each line.
            const stripPrefix = new RegExp('^ {0,' + indentation.length + '}');
            contents = contents.split('\n').map((line) => line.replace(stripPrefix, '')).join('\n');
          }
          contents = escapeCharacters(
            contents,
            ' \t*_{}[]()<>#+=.!|`-',
            /*afterBackslash=*/ false,
            /*doNotEscapeTildeAnDollar=*/ true,
          );
```

para evitar tener dos llamadas a `escapeCharacters`.